source/base/mpi.cc

```diff
@@ -511,9 +511,82 @@ namespace Utilities
       const MPI_Comm &                 mpi_comm,
       const std::vector<unsigned int> &destinations)
     {
-      return compute_point_to_point_communication_pattern(mpi_comm,
-                                                          destinations)
-        .size();
+      auto destinations_unique = destinations;
+      std::sort(destinations_unique.begin(), destinations_unique.end());
+      destinations_unique.erase(std::unique(destinations_unique.begin(),
+                                            destinations_unique.end()),
+                                destinations_unique.end());
+
+      if (Utilities::MPI::min<unsigned int>(destinations.size() ==
```

Comment:
can you add a comment what the `if` actually checks? Maybe "If the list of destinations did not include any duplicates, then ..."